okonomiyaki/file_formats/_egg_info.py

```diff
@@ -0,0 +1,697 @@
+
+    def _to_dict(self):
+        raw_data = {
+            _TAG_NAME: self.name,
+            _TAG_VERSION: self.version,
+            _TAG_BUILD: self.build,
+            _TAG_ARCH: self.arch,
+            _TAG_PLATFORM: self.platform,
+            _TAG_OSDIST: self.osdist,
+            _TAG_PACKAGES: [str(p) for p in self.packages],
+            _TAG_PYTHON: self.python,
+            _TAG_PYTHON_TAG: self.python_tag,
+            _TAG_METADATA_VERSION: self.metadata_version
+        }
+
+        ret = {}
+        for k, v in raw_data.items():
+            if isinstance(v, six.string_types):
+                v = str(v)
+            ret[k] = v
+        return ret
+
+    def to_string(self):
+        """
+        Returns a string that is suitable for the depend file inside our
+        legacy egg.
+        """
+        template = _METADATA_TEMPLATES.get(self.metadata_version, None)
+        data = self._to_dict()
+
+        # This is just to ensure the exact same string as the produced by the
+        # legacy buildsystem
+        if len(self.packages) == 0:
+            data[_TAG_PACKAGES] = "[]"
+        else:
+            data[_TAG_PACKAGES] = "[\n{0}\n]". \
+                format("\n".join("  '{0}',".format(p)
+                       for p in self.packages))
+        return template.format(**data)
+
+
+class LegacySpec(HasTraits):
+    """
+    This models the EGG-INFO/spec content.
+    """
+    depend = Instance(LegacySpecDepend)
+    """
+    Models the spec/depend content
+    """
+
+    lib_depend = List()
+    """
+    List of freeform content
+    """
+    lib_provide = List()
+    """
+    List of freeform content
+    """
+
+    summary = Unicode()
+    """
+    Summary metadata of the egg.
+    """
+
+    @classmethod
+    def from_egg(cls, egg):
+        spec_depend = LegacySpecDepend.from_egg(egg)
+
+        data = {"depend": spec_depend}
+
+        with zipfile2.ZipFile(egg) as fp:
+            try:
+                lib_depend_data = fp.read(_SPEC_LIB_DEPEND_LOCATION).decode()
+                data["lib_depend"] = lib_depend_data.splitlines()
+            except KeyError:
+                pass
+
+        return cls(**data)
+
+    @property
+    def egg_name(self):
+        return "{0}-{1}-{2}.egg".format(self.depend.name,
+                                        self.depend.version,
+                                        self.depend.build)
+
+    def depend_content(self):
+        return self.depend.to_string()
+
+    def lib_depend_content(self):
+        """
+        Returns a string that is suitable for the lib-depend file inside our
+        legacy egg.
+        """
+        # The added "" is for round-tripping with the current egg format
+        return "\n".join(str(entry) for entry in self.lib_depend + [""])
+
+
+Dependencies = collections.namedtuple("Dependencies", ["build", "runtime"])
+
+_TAG_RE = re.compile("(?P<interpreter>(cp|pp|cpython))(?P<version>([\d_]+))")
```

Comment:
I can understand omitting `jy` and `ip`, but I think we have to at least include the generic `py`.